padertorch/contrib/examples/vae_mnist/mnist_vae_example.py

```diff
@@ -0,0 +1,195 @@
+"""
+A very simple training example for a Variational Autoencoder(VAE)
+using the mnist database, sacred and the padertorch trainer
+
+Use the following command:
+    python mnist_vae_example.py with logfilepath=/path/to/dir/ epochs=500
+
+Other options for sacred are available, refer to the config() function.
+"""
+import os
+
+import padertorch
+import padertorch.train.optimizer as pt_opt
+
+import sacred
+
+import torch
+import torch.nn as nn
+import torchvision
+import torchvision.transforms as transforms
+
+from einops import rearrange
+
+ex = sacred.Experiment()
+
+
+def get_dataset(batch_size=128):
+    """
+    load MNIST dataset and store it in ./data
+
+    Args:
+        batch_size (int):
+
+    Returns:
+        trainloader and testloader
+
+    """
+    tf = transforms.Compose([transforms.ToTensor()])
+
+    # Get Datasets
+    trainset = torchvision.datasets.MNIST(root='./data', train=True,
+                                          download=True, transform=tf)
+    trainloader = torch.utils.data.DataLoader(trainset, batch_size=batch_size,
+                                              shuffle=True, num_workers=2)
+
+    testset = torchvision.datasets.MNIST(root='./data', train=False,
+                                         download=True, transform=tf)
+    testloader = torch.utils.data.DataLoader(testset, batch_size=batch_size,
+                                             shuffle=False, num_workers=2)
+
+    return trainloader, testloader
+
+
+class PadertorchModel(padertorch.base.Model):
+    MSE_loss = nn.MSELoss()
+
+    def __init__(self, net):
+        super().__init__()
+        self.net = net
+
+    def forward(self, inputs):
+        images = inputs[0]
+        out, mu, std = self.net(images)
+
+        return dict(
+            prediction=out,
+            mean=mu,
+            std=std
+        )
+
+    def review(self, inputs, outputs):
+        images = rearrange(inputs[0], 'b c x y -> b (c x y)')
+        prediction = rearrange(outputs['prediction'], 'b c x y -> b (c x y)')
+
+        MSE = self.MSE_loss(prediction, images)  # reconstruction error
+        # Kullback-Leibler divergence
+        KLD = -0.5 * torch.sum(1 + outputs['std'] - outputs['mean'].pow(2) - outputs['std'].exp())
```

Comment:
there is already a preimplemented version of the KL-divergence in padertorch.ops.losses